db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/Scheduler.java

```diff
@@ -354,8 +357,31 @@ public List<CurrentlyExecuting> getCurrentlyExecutingWithStaleHeartbeat() {
   @SuppressWarnings({"rawtypes", "unchecked"})
   protected void detectDeadExecutions() {
     LOG.debug("Deleting executions with unresolved tasks.");
-    taskResolver
-        .getUnresolvedTaskNames(deleteUnresolvedAfter)
+
+    Map<String, Instant> taskToNewestExecutionTime = new HashMap<>();
+    schedulerTaskRepository.getScheduledExecutions(
+        ScheduledExecutionsFilter.all(),
+        execution ->
+            taskToNewestExecutionTime.merge(
+                execution.taskInstance.getTaskName(),
+                execution.executionTime,
+                (oldValue, newValue) -> oldValue.isAfter(newValue) ? oldValue : newValue));
+
```

Comment:
I would rather avoid this...

Is there perhaps a way to have the TaskResolver also get information on how long the task-name has been unresolved 🤔 (and use that, rather than fetching all executions into mem and checking)

Reply:
I.e. do something [here](https://github.com/kagkarlsson/db-scheduler/blob/master/db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/jdbc/JdbcTaskRepository.java#L792):

Using the execution-time (`Instant executionTime = jdbcCustomization.getInstant(rs, "execution_time");`)